src/Microsoft.AspNetCore.OData/Adapters/WebApiActionDescriptor.cs

```diff
@@ -112,7 +112,7 @@ public string ActionName
         /// <returns>A list of attributes of type T.</returns>
         public IEnumerable<T> GetCustomAttributes<T>(bool inherit) where T : Attribute
         {
-            return this.innerDescriptor.ControllerTypeInfo.GetCustomAttributes<T>(inherit);
+            return this.innerDescriptor.MethodInfo.GetCustomAttributes<T>(inherit);
```

Comment:
@habbes Can you explain this change?

Reply:
@habbes could this be a set of maybe controller and the method attributes so as to retain the old behaviour?

Reply:
This method is supposed to return the custom attributes of the underlying action descriptor. In the case of AspNetCore the underlying descriptor is a `ControllerActionDescriptor`. How it was returning the attributes of the controller type, rather than of the action method. I believe that was a bug, since the `WebApiActionDescriptor` is a wrapper around the action descriptor of an action method, not the controller. So I changed it to return the attributes of the method instead.

Reply:
@marabooy, my suspicion is that the old behaviour was a bug. @xuzhg  could you comment on this?